src/components/notifications/AppNotifications/AppNotificationItem.tsx

```diff
@@ -16,34 +22,131 @@ interface IAppNotificationProps {
   appLogo: string
 }
 
+const DRAG_OFFSET = 150
+
 const AppNotificationItem: React.FC<IAppNotificationProps> = ({ notification, appLogo }) => {
+  const [notificationsDrag, setNotificationsDrag] = useContext<AppNotificationsDragContext>(
+    AppNotificationDragContext
+  )
   const formattedTime = useFormattedTime(notification.timestamp)
   const [dropdownToShow, setDropdownToShow] = useState<string | undefined>()
 
+  const dragControls = useAnimationControls()
+  const actionControls = useAnimationControls()
+
+  useEffect(() => {
+    // If current notification is dragged, hide dragControls for all other notifications
+    const currentNotificationItem = notificationsDrag?.find(
+      n => n.id.toString() === notification.id
+    )
+    if (!currentNotificationItem?.isDragged) {
+      dragControls.start('hidden')
+      actionControls.start('hidden')
+    }
+  }, [notificationsDrag])
+
+  // On clicking a button, it should reset dragControls to hidden
+  const handleClearClick = () => {
+    // Do more stuff here
+    dragControls.start('hidden')
+    actionControls.start('hidden')
+  }
+
+  const handleUnreadClick = () => {
+    // Do more stuff here
+    dragControls.start('hidden')
+    actionControls.start('hidden')
+  }
+
+  const handleDragEnd = (_: never, info: PanInfo) => {
+    const shouldShowControls = info.velocity.x >= 50 || info.offset.x > DRAG_OFFSET / 2
+    if (shouldShowControls) {
+      dragControls.start('visible')
+    } else {
+      dragControls.start('hidden')
+    }
+  }
+
+  const handleDrag = (_: never, info: PanInfo) => {
+    const shouldShowActions = info.velocity.x >= 0 || info.offset.x > DRAG_OFFSET / 2
+    if (shouldShowActions) {
+      actionControls.start('visible')
+    } else {
+      actionControls.start('hidden')
+    }
+  }
+
+  // Change current items drag state to true and all others to false
+  const handleDragStart = () => {
+    setNotificationsDrag(
```

Comment:
It might be safer to use the currentState:

```ts
setNotificationsDrag(currentNotifications =>
      currentNotifications?.map(n => {
        return {
          ...n,
          isDragged: n.id.toString() === notification.id
        }
      })
    )
```

Reply:
Ah! Perfect, ill get that goin